vyper/context/types/utils.py

```diff
@@ -13,13 +14,65 @@
     StructureException,
     UndeclaredDefinition,
     UnknownType,
+    VyperInternalException,
 )
 
 
+class StringEnum(enum.Enum):
+    @staticmethod
+    def auto():
+        return enum.auto()
+
+    # Must be first, or else won't work, specifies what .value is
+    def _generate_next_value_(name, start, count, last_values):
+        return name.lower()
+
+    # Override ValueError with our own internal exception
+    @classmethod
+    def _missing_(cls, value):
+        raise VyperInternalException(f"{value} is not a valid {cls.__name__}")
+
+    @classmethod
+    def is_valid_value(cls, value: str) -> bool:
+        return value in set(o.value for o in cls)
+
+    @classmethod
+    def options(cls) -> List["StringEnum"]:
+        return list(cls)
+
+    @classmethod
+    def values(cls) -> List[str]:
+        return [v.value for v in cls.options()]
+
+    # Comparison operations
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, self.__class__):
+            super().__eq__(other)
+        return self is other
+
+    def __ne__(self, other: object) -> bool:
```

Comment:
This isn't needed, python will automatically call to `__eq__` and invert the result.